### ISSUE TITLE
Wire proto schema generation broken for protobuf messages containing a map with "protobuf.Any" type fields on versions >= 4.9.4

### ISSUE THREAD

**Issue by abhisheksharma-dd** [square/wire#2862]
Looks like https://github.com/square/wire/pull/2797 introduced a bug where wire proto schema generation is broken for a protobuf message containing a map field whose value is of type protobuf.Any. See error message:

> _redacted.proto_ needs to import google/protobuf/any.proto
    for field additional_info (/Users/abhishek.sharma/Projects/.../redacted.proto:1294:5)
    in message AgencyRenderDetails (/Users/abhishek.sharma/Projects/.../redacted.proto:1287:3)

This was working for versions <= 4.9.3 and the PR was only introduced in v4.9.4. To isolate the issue to that PR, we also cloned the repo and tested v4.9.7 with just the PR(and a follow up fix: https://github.com/square/wire/pull/2800) reverted and things appear to work again.

Here's a small protobuf file that should be able to trigger the bug:
```
syntax = "proto3";

package squareup.dinosaurs;

option java_package = "com.squareup.dinosaurs";

import "google/protobuf/any.proto";
import "google/protobuf/wrappers.proto";

message Dinosaur {
  // Common name of this dinosaur, like "Stegosaurus".
  google.protobuf.StringValue name = 1;

  // URLs with images of this dinosaur.
  map<string, google.protobuf.Any> string picture_urls = 2;
}
```
The generated proto file by wire will not contain the `google/protobuf/any.proto` import thus failing the `generateProtos` gradle task. Note that this most likely has nothing to do with specifically the Any type, more generally if one needs to define a map using a type defined in another proto file, they will not be able to do for any version >= 4.9.4 

While a root cause is being figured out, can the change be reverted please?

**Comment by oldergod** [comment]
Thanks for reporting. A fix is happening in #2864

**Comment by abhisheksharma-dd** [comment]
Thank you. Do you have an ETA for cutting a new release with the fix?

**Comment by abhisheksharma-dd** [comment]
Hi @oldergod thank you for merging the fix. Could we also get a new version the with fix released?

**Comment by oldergod** [comment]
released in 4.9.8
